sdk/core/core/src/az_aad_private.h

```diff
@@ -38,6 +38,22 @@ AZ_NODISCARD bool _az_token_expired(_az_token const* token);
 
 AZ_NODISCARD az_result _az_token_set(_az_token* self, _az_token const* new_token);
 
+/**
```

Comment:
This is only here for unit testing, right?
Maybe add a comment to that effect.
Or, you could remove this from here and just copy the signature directly into the unit test file (test_az_encode.c).

Reply:
New set of compiler warnings would not let me to define non-static (linker exported) function without prior declaration. So in order to remove it from header (which is already private BTW), I'll have to declare it in a C file, and then define it. I don't think we should do that. Plus, this way there is bigger chance that signatures will mismatch. I see sometimes it could be worth doing, but not in this case IMO.